### ISSUE TITLE
feature request: a browse button for folder paths

### ISSUE THREAD

**Issue by marrie** [syncthing/syncthing#2958]
Reason: typing in a very  long path is not fun. This would be handy for filling  in the path of a folder in case the path is extremely deep in the folder structure. This would also help newbies as well as they might not know what a path is. I get complaints from people saying this program is too hard and that I should email them let;let's say 15 gig zip files. That won't happen any time soon.

Thanks, and really love the program.

**Comment by AudriusButkevicius** [comment]
Browsers do not support that.

**Comment by AudriusButkevicius** [comment]
Also, check community contributions, (such as synctrayzor for Windows) which help deliver the experience you are looking for.

**Comment by marrie** [comment]
Oh really? I thought they did, as I see brows buttons on applications such as google photoes and youtube. I can brows for the file instead of entering the path it manually. I also use a brows button on wordpress to add files to the library if needed. I think this is app specific not browser specific.

**Comment by marrie** [comment]
I'm on a mac. So that won't help at all here. I know my beginning users are on windows but they don't want to use stuff like that for some reason. I don't want to exclude them hence why I suggested a brows butto, more for them then me lol!

**Comment by AudriusButkevicius** [comment]
You can browse for files, not for directories.

**Comment by marrie** [comment]
Oh duh! Ok, I guess I need more coffee. Oh well, looks like I need to find a way to copy pathes in what ever platform I'm in. Path copy  works under windows, OSX is a bit tricky and not sure about linux as I don't have syncthing on there as of yet.

Sorry about that. 

Take care and have a happy Friday.

**Comment by wweich** [comment]
The "Browse" buttons in browser are only for file uploads. There is no way for the side to see the path, only filename and contents.

Also: Syncthing has auto complete for the folder path. Typing `/` should give you a dropdown list with all root folders and you can go from there. `~` is immediately replaced with you home folder.

**Comment by lkwg82** [comment]
I think Marie is right. A folder dialog is needed. I think it will go with an API endpoint and a handcrafted Dialog. I misses it too.

**Comment by canton7** [comment]
Wasn't there some work towards a file browser built with javascript some time ago? I remember being told about it around a year ago...

**Comment by AudriusButkevicius** [comment]
For ignores yeah, not for general OS browsing.

**Comment by canton7** [comment]
Hrmph I can't find an issue link, just this note on one of my issues 

> @calmh mentioned the possibility of putting a more fully-fledged folder browser in here

**Comment by AudriusButkevicius** [comment]
https://github.com/syncthing/syncthing/tree/selective

**Comment by lkwg82** [comment]
@AudriusButkevicius Can you summarize your progress in dones and todos? So someone is able to catch up.

**Comment by AudriusButkevicius** [comment]
So that branch there was todo with something else.
It was todo with providing a directory tree to select which folders you want synced or ignored.
The tree was there, construction of patterns from the tree was there, the only thing was missing was actually ignoring stuff that was not selected in the tree.

**Comment by AudriusButkevicius** [comment]
I think that does not fit in with this.

It's probably easier to take some directory browser javascript library, and wedge in some API endpoint to serve it to satisfy this feature.

**Comment by lkwg82** [comment]
These information saved a lot of time. Is the branch in general outdated and can be removed?

**Comment by AudriusButkevicius** [comment]
Well it contains many hours of my work, and it's fairly close to being a feature.

I think it will be useful with the ignore rework that might happen some day.

**Comment by lkwg82** [comment]
If you say it is close to being complete and on the other hand suggest it is better to use another approach I'm confused. What made you not complete it yet?

**Comment by AudriusButkevicius** [comment]
It's a completely different feature, nothing todo with a browse button described in this ticket. It just happens to provide a limited functionality of a folder browser.

**Comment by calmh** [comment]
The API endpoint is already there though in the form of /rest/system/browse.

**Comment by AudriusButkevicius** [comment]
It doesn't work outside of the the folder root.
We are talking about providing a filesystem browser at the point you create a folder.

**Comment by wweich** [comment]
Issuing a GET request to `/rest/system/browse?current=%2F`does result in the list of folders in my file system root (`/`).
This API endpoint is used by the auto complete for the folder path.

**Comment by AudriusButkevicius** [comment]
Sorry, I mixed it up a different endpoint.

It still needs special treatment for Windows, listing available drives.

**Comment by bugith** [comment]
Hi Marrie, I wrote in the [syncthing forum a kind of tuto](https://forum.syncthing.net/t/add-folder-folder-path-tip-for-windows-users/7195) "Share folder-Folder Path : Tip for Windows user" that as a paragraph that applies to many OSes, many Apps, many Browsers. I hope users really interested in taming their keyboard-machines will gain big benefit in trying it step by step. After all it seems to be only a matter of a very small subset of keyboard shortcuts and special keys. Unplug the mouse

**Comment by canton7** [comment]
I should also add: SyncTrayzor for windows includes a Browse button, which opens the native folder browser.

**Comment by marrie** [comment]
Yah  in my case I'm working outside of my home directory. I'm familiar with linux and unix so I already knew all of this. However a beginner might not know how to work outside of their home directory. Hence a bit of a more friendly way to work with folder paths in syncthing.

**Comment by Getron** [comment]
It would be great to have one.

Maybe we can use this one, or a part of it:
https://github.com/simogeo/Filemanager

It is a php file manager, and can be implemented, i think.

Any other suggestions?

**Comment by canton7** [comment]
jstree, too.

**Comment by canton7** [comment]
Works in Chrome, at least: https://whatpwacando.today/file-system. Although you do get lovely permission prompts, as it wants to give the browser permission to read everything in the folder as well.

**Comment by bt90** [comment]
We could add a REST endpoint for path completion, but IMHO that's akin to using a sledgehammer to crack a nut.
